### PR TITLE
SSCS-10016 add isConfidential as filter

### PR DESCRIPTION
**Pull request by pk2453**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by ikirsanov — APPROVED** [review]

**Review by dattipoeHMCTS — COMMENTED** [review]
Do you need the SearchResult field too? Don't forget version bumps

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json [hunk 1]
```diff
@@ -10,12 +10,13 @@
...
-  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "languagePreferenceWelsh", "Label": "Welsh language flag", "DisplayOrder": 13},
-  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "translationWorkOutstanding", "Label": "Translation outstanding", "DisplayOrder": 14},
-  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "reasonableAdjustmentsOutstanding", "Label": "Alternative letter format required", "DisplayOrder": 15},
-  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "hasUnprocessedAudioVideoEvidence", "Label": "Does case have unprocessed audio/video evidence?", "DisplayOrder": 16},
-  {"LiveFrom": "16/12/2021", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "dwpDueDate", "Label": "FTA response due date", "DisplayOrder": 17},
+  {"LiveFrom": "17/08/2022", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "isConfidentialCase", "Label": "Is a Confidential Case?", "DisplayOrder": 12},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:13]: Does this field (and the rest) need to be behind a feature flag?

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json [hunk 2]
```diff
@@ -10,12 +10,13 @@
...
+  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "isScottishCase", "Label": "Is a Scottish case?", "DisplayOrder": 13},
+  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "languagePreferenceWelsh", "Label": "Welsh language flag", "DisplayOrder": 14},
+  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "translationWorkOutstanding", "Label": "Translation outstanding", "DisplayOrder": 15},
+  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "reasonableAdjustmentsOutstanding", "Label": "Alternative letter format required", "DisplayOrder": 16},
+  {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "hasUnprocessedAudioVideoEvidence", "Label": "Does case have unprocessed audio/video evidence?", "DisplayOrder": 17},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: Don't forget to modify the ordering of the nonprod sheets too for these

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json (hunk 1 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:13]: Haven't seen any such requirement

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json (hunk 2 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: This field is only added in prod one and not in non-prod so ordering in non-prod will remain same.

**Comment by pk2453** [comment]
> Do you need the SearchResult field too? Don't forget version bumps

Not sure what is meant by "need the SearchResult field too". JUst need to ability to filter the cases based on if they are confidential.

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json (hunk 2 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: It does, you need to change the DisplayOrder on the nonprod sheets too. For example, for caseworker-sscs-superuser, dwpDueDate is now the same order as dwpIsOfficerAttending in the nonprod sheet. 
You need to increment the DisplayOrder on the non prod sheets too

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json (hunk 1 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:13]: Can you enquire about it then to make sure?

**Review by TeaWithLucas — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json (hunk 2 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: @TeaWithLucas  does that mean I also need to add that new field on non-prod as well?

Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: > It does, you need to change the DisplayOrder on the nonprod sheets too. For example, for caseworker-sscs-superuser, dwpDueDate is now the same order as dwpIsOfficerAttending in the nonprod sheet. You need to increment the DisplayOrder on the non prod sheets too

updated non-prod ordering as well

Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: > @TeaWithLucas does that mean I also need to add that new field on non-prod as well?

No, prod fields are added combined with nonprod sheets outside of production

Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: > > It does, you need to change the DisplayOrder on the nonprod sheets too. For example, for caseworker-sscs-superuser, dwpDueDate is now the same order as dwpIsOfficerAttending in the nonprod sheet. You need to increment the DisplayOrder on the non prod sheets too
> 
> updated non-prod ordering as well

Can you update WorkBasketInputFields-postponements-nonprod.json too please

Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields.json:18]: updated this as well

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 [
   {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-dwpresponsewriter", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 10},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:2]: Needs bumping

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [
   {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "unprocessedPostponementRequest", "Label": "Postponement request", "DisplayOrder": 9},
   {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-registrar", "CaseFieldID": "unprocessedPostponementRequest", "Label": "Postponement request", "DisplayOrder": 9},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:3]: Needs bumping

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [
   {"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "unprocessedPostponementRequest", "Label": "Postponement request", "DisplayOrder": 9},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:2]: Needs bumping

**Review by TeaWithLucas — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json (hunk 4 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:3]: anything below 11 should not be needed bump as new one added with order 12

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json (hunk 5 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:2]: anything below 11 should not be needed bump as new one added with order 12

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json (hunk 3 above)
Comment (pk2453) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:2]: anything below 11 should not be needed bump as new one added with order 12

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json (hunk 4 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:3]: Conflicts with
{"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-registrar", "CaseFieldID": "hasUnprocessedAudioVideoEvidence", "Label": "Does case have unprocessed audio/video evidence?", "DisplayOrder": 9},

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json (hunk 5 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-postponements-nonprod.json:2]: Conflicts with
{"LiveFrom": "20/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "hasUnprocessedAudioVideoEvidence", "Label": "Does case have unprocessed audio/video evidence?", "DisplayOrder": 9},

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json (hunk 3 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:2]: conflicts with:
{"LiveFrom": "18/06/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-dwpresponsewriter", "CaseFieldID": "createdInGapsFrom", "Label": "Is a new digital case?", "DisplayOrder": 10},

**Review by TeaWithLucas — CHANGES_REQUESTED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Review by EliandroGV1 — APPROVED** [review]

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-dwpresponsewriter", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 11},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-clerk", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 19},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-judge", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 10},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-registrar", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 12},
+  {"LiveFrom": "10/10/2019", "CaseTypeID": "Benefit", "UserRole": "caseworker-sscs-superuser", "CaseFieldID": "dwpIsOfficerAttending", "Label": "Is the Presenting officer attending", "DisplayOrder": 18},
```
Comment (TeaWithLucas) [inline benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:6]: Why the removal of the captial O but not the capital P?
I think I prefer it with a capital O as it is a title.
The '?' should stay.

Comment (hgulv1) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:6]: PO approved this change already

**Review by TeaWithLucas — DISMISSED** [review]

File: benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json (hunk 6 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/WorkBasketInputFields/WorkBasketInputFields-sandl-nonprod.json:6]: Remember they don't look at the PRs so don't know to look for changes that aren't on the ticket. I will approve but I recommend you confirm this with a BA first.

**Review by TeaWithLucas — APPROVED** [review]

**Review by hajam09 — APPROVED** [review]

**Pull request merged**